components/observations/ObservationImageEditView.tsx

```diff
@@ -0,0 +1,421 @@
+  // all the way
+  const finalHeight = useRef(INITIAL_HEIGHT);
+
+  const onLayout = useCallback(
+    (event: LayoutChangeEvent) => {
+      // If the layout is resized while the view is dismissing (e.g. the keyboard is going away)
+      // just set them minimum view size to 0 because we want it to animate all the way out
+      if (state.mode === 'dismissing') {
+        actions.setViewBoundary({min: -edges.bottom});
+        return;
+      }
+
+      // Calculate the max height base on the safe area
+      const height = event.nativeEvent.layout.height;
+      finalHeight.current = height;
+      const maxHeight = height - edges.top;
+      actions.setViewBoundary({max: maxHeight});
+    },
+    [edges.top, state.mode, actions, edges.bottom],
+  );
+
+  // Called when the panResponder completes, if it's dragged far enough or fast enough
+  // it will request the view to be dismissed
+  const onDragDismiss = (velocity?: number) => {
+    if (velocity != null && velocity > AUTO_DISMISS_VELOCITY) {
+      actions.dismissRequest();
+      return;
+    }
+    ref.current?.measure((_x, _y, _w, h) => {
+      if (h < AUTO_DISMISS_DRAGGING_HEIGHT) {
+        actions.dismissRequest();
+      }
+    });
+  };
+
+  // listen to dragging gestures and set the dragRef value to move the inter view
+  const panResponder = useRef(
+    PanResponder.create({
+      onMoveShouldSetPanResponder: () => true,
+      onStartShouldSetPanResponder: () => true,
+      onPanResponderMove: Animated.event([null, {dy: dragRef.current}], {useNativeDriver: false}),
+      onPanResponderRelease: (_, gestureState) => {
+        onDragDismiss(gestureState.vy);
+        Animated.decay(dragRef.current, {
+          velocity: gestureState.vy,
+          deceleration: 0.98,
+          useNativeDriver: false,
+        }).start(() => {
+          dragRef.current.extractOffset();
+          onDragDismiss();
+        });
+      },
+    }),
+  ).current;
+
+  // validate the form and dismiss if autoDismiss is true which is the default
+  const onSubmitPress = useCallback(() => {
+    void (async () => {
+      await formContext.trigger();
+
+      const submit = formContext.handleSubmit(values => {
+        onSetCaption(values.caption);
+      });
+
+      await submit();
+
+      if (autoDismiss && state.mode !== 'dismissing') {
+        actions.dismissRequest();
+      }
+    })();
+  }, [formContext, onSetCaption, autoDismiss, actions, state.mode]);
+
+  const onPressOverlay = useCallback(() => {
+    actions.dismissRequest();
+  }, [actions]);
+
+  const transitionAnimationRef = useRef<Animated.CompositeAnimation | null>(null);
+
+  useEffect(() => {
+    if (state.mode !== 'dismissRequested') {
+      return;
+    }
+    actions.dismiss();
+
+    transitionAnimationRef.current?.stop();
+    textInputRef.current?.blur();
+    transitionAnimationRef.current = Animated.parallel([
+      Animated.timing(fadeAnim.current, {
+        toValue: 0,
+        duration: 400,
```

Comment:
I'm not sure if this is indicative of what users will experience, but once I accidentally opened the modal and tried to close it before the animation had ended, which caused the input to get lost forever:

```
 LOG   >> ACTION {"action": "presentRequest"}
 LOG   >>  {"keyboardVisible": false, "mode": "resting", "viewBoundary": {"max": 0, "min": 0}}
 LOG   <<  {"keyboardVisible": false, "mode": "presentRequested", "mutatedAt": 1712235746648, "previousMode": "resting", "viewBoundary": {"max": 0, "min": 0}}
 LOG   >> ACTION {"action": "present"}
 LOG   >>  {"keyboardVisible": false, "mode": "presentRequested", "mutatedAt": 1712235746648, "previousMode": "resting", "viewBoundary": {"max": 0, "min": 0}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746660, "previousMode": "presentRequested", "viewBoundary": {"max": 0, "min": 0}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"max": 797}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746660, "previousMode": "presentRequested", "viewBoundary": {"max": 0, "min": 0}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746672, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 0}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 32}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746672, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 0}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746680, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 32}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 54}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746680, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 32}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746688, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 54}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"max": 461}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746688, "previousMode": "presentRequested", "viewBoundary": {"max": 797, "min": 54}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746729, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 53.9998779296875}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746729, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746818, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 53.9998779296875}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 54.0001220703125}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746818, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 53.9998779296875}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746835, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54.0001220703125}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 53.9998779296875}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746835, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54.0001220703125}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746868, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 53.9998779296875}}
 LOG   >> ACTION {"action": "setBoundary", "minMax": {"min": 54}}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746868, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 53.9998779296875}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746885, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   >> ACTION {"action": "dismissRequest"}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746885, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   <<  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746885, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   >> ACTION {"action": "presentCompleted"}
 LOG   >>  {"keyboardVisible": false, "mode": "presenting", "mutatedAt": 1712235746885, "previousMode": "presentRequested", "viewBoundary": {"max": 461, "min": 54}}
 LOG   <<  {"keyboardVisible": false, "mode": "presented", "mutatedAt": 1712235747671, "previousMode": "presenting", "viewBoundary": {"max": 461, "min": 54}}
```